client/state/sites/selectors.js

```diff
@@ -20,17 +20,46 @@ import {
 import createSelector from 'lib/create-selector';
 import { fromApi as seoTitleFromApi } from 'components/seo/meta-title-editor/mappings';
 import versionCompare from 'lib/version-compare';
+import getComputedAttributes from 'lib/site/computed-attributes';
 
 /**
- * Returns a site object by its ID.
+ * Returns a raw site object by its ID.
  *
  * @param  {Object}  state  Global state tree
  * @param  {Number}  siteId Site ID
  * @return {?Object}        Site object
  */
-export function getSite( state, siteId ) {
+export const getRawSite = ( state, siteId ) => {
 	return state.sites.items[ siteId ] || null;
-}
+};
+
+/**
+ * Returns a normalized site object by its ID. Intends to replicate
+ * the site object returned from the legacy `sites-list` module.
+ *
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {?Object}        Site object
+ */
+export const getSite = createSelector(
+	( state, siteId ) => {
+		const site = getRawSite( state, siteId );
+
+		if ( ! site ) {
+			return null;
+		}
+
+		return {
+			...site,
+			...getComputedAttributes( site ),
```

Comment:
My suspicions started tingling that `getComputedAttributes` could have been mutating state here. It turned out not to be true (i.e. everything's fine!) but maybe a case for using `deepFreeze` in selector testing.